console/views.py

```diff
@@ -65,3 +71,53 @@ def decrypt():
 
     else:
         return render_template('decrypt.html')
+
+
+def get_all_responses():
+    data = SurveyResponse.query.all()
+    return data
+
+
+def get_store_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest):
+    responses = get_all_responses()
+    store_data = []
+    for response in responses:
+        store_data.append(response.data)
+
+    filtered_data = []
+    for data in store_data:
+        if (tx_id == '') or (data['tx_id'] == tx_id):
+            if (ru_ref == '') or (data['metadata']['ru_ref'] == ru_ref):
+                if (survey_id == '') or (data['survey_id'] == survey_id):
+                    filtered_data.append(data)
```

Comment:
Why aren't you doing the heavy lifting here with SQL Alchemy? This should be done on the SQL end, not in memory in the app. See http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.filter_by for usage.